pkg/ipmi/ipmi.go

```diff
@@ -173,3 +188,61 @@ func (i *IPMI) ShutoffWatchdog() error {
 
 	return nil
 }
+
+func (i *IPMI) setsysinfo(data *setSystemInfoReq, dataLen uint16) error {
+	req := &req{}
+	req.msg.cmd = _SET_SYSTEM_INFO_PARAMETERS
+	req.msg.netfn = _IPMI_NETFN_APP
+	req.msg.dataLen = dataLen
```

Comment:
How about making this one a fixed size value of 18 bytes?
* The spec doesn't mention this is variable size. It mentions: `data 2:17 - 16-byte block for system firmware name string data`
* The `ipmitool` also has a fixed size value for this:
https://github.com/ipmitool/ipmitool/blob/master/lib/ipmi_mc.c#L1576

So, how about changing/simplifying this function to: `func (i *IPMI) setsysinfo(data *setSystemInfoReq) error`.

The string length is already set in line 226: `data.strData[1] = byte(len)`.

Reply:
The spec doesn't say it cannot be a variable size either, when I use string length shorter than 14 bytes it can still work and no IPMI error so I think it supports variable size. Following ipmitool that always send 16 bytes of string (zero-padded if the string is shorter) can also work, so I updated the commit to follow it.